sql/distinct.go

```diff
@@ -37,6 +37,9 @@ func (*planner) distinct(n *parser.Select, p planNode) planNode {
 		d.columnsInOrder = make([]bool, len(p.Columns()))
 	}
 	for _, p := range p.Ordering() {
+		if p == 0 {
```

Comment:
The layers of plans each exporting a public interface, and at the same time, that interface being used internally makes this code very hard to reason about.